src/cli/self_update/windows.rs

```diff
@@ -929,57 +912,79 @@ mod tests {
 
     #[test]
     fn windows_path_regkey_type() {
-        // per issue #261, setting PATH should use REG_EXPAND_SZ.
         let _guard = RegistryGuard::new(&USER_PATH);
-        let root = RegKey::predef(HKEY_CURRENT_USER);
-        let environment = root
-            .open_subkey_with_flags("Environment", KEY_READ | KEY_WRITE)
-            .unwrap();
-        environment.delete_value("PATH").unwrap();
+
+        let environment = CURRENT_USER.create("Environment").unwrap();
+        environment.remove_value("PATH").unwrap();
 
         {
             // Can't compare the Results as Eq isn't derived; thanks error-chain.
             #![allow(clippy::unit_cmp)]
             assert_eq!((), super::_apply_new_path(Some(wide("foo"))).unwrap());
         }
-        let root = RegKey::predef(HKEY_CURRENT_USER);
-        let environment = root
-            .open_subkey_with_flags("Environment", KEY_READ | KEY_WRITE)
-            .unwrap();
-        let path = environment.get_raw_value("PATH").unwrap();
-        assert_eq!(path.vtype, RegType::REG_EXPAND_SZ);
-        assert_eq!(super::to_winreg_bytes(wide("foo")), &path.bytes[..]);
+
+        let environment = CURRENT_USER.create("Environment").unwrap();
+        let path = environment.get_hstring("PATH").unwrap();
+
+        fn pcwstr<T: AsRef<str>>(value: T) -> Vec<u16> {
```

Comment:
This function was peeked from `windows-registry`. Maybe it should be removed and use `w!()` macro instead.

Reply:
> Yes, that would avoid a heap allocation.

I tested but I cannot replace this function with `w!()`, because macros like `w!()` only accepts string literals, while this function accepts both literals and variables.

In addition, I tried modifying `w!()` macro to make it accepts variables, but failed to avoid heap allocations.😵